KoPrivateGPT/benchmark/retrieval/strategyQA.py

```diff
@@ -0,0 +1,33 @@
+def strategyQA(solution: dict, pred: dict, metrics_factories: list, k_value: int):
+    '''
+    k shuld be smaller than the number of paragraphs in the prediction
+    '''
+    final_score = {f'{metric_factory.__name__[:-7]}@{str(k_value)}': list() for metric_factory in metrics_factories}
```

Comment:
Why don't you make get_type() function at BaseRetrievalEvaluationFactory as member method other than metric_factory.__name__[:-7]. Naming policy is so mutable.

Reply:
Ummmm... nice idea. I'm gonna think about BaseRetrievalEvalFac member for some thinkable situation, use case

Reply:
use metric_name property